core-bundle/src/Controller/AbstractFragmentController.php

```diff
@@ -75,47 +83,91 @@ protected function getPageModel(): ?PageModel
     }
 
     /**
-     * Creates a template by name or from the "customTpl" field of the model.
+     * Creates a FragmentTemplate container object by template name or from the
+     * "customTpl" field of the model and registers the effective template as
+     * default view when using render().
+     *
+     * Calling getResponse() method on the returned object will internally call
+     * render() with the set parameters and return the response.
+     *
+     * Note: The $fallbackTemplateName argument will be removed in Contao 6;
+     * always set a template via the fragment options, instead.
      */
-    protected function createTemplate(Model $model, string $templateName): Template
+    protected function createTemplate(Model $model, string|null $fallbackTemplateName = null): FragmentTemplate
     {
-        if (isset($this->options['template'])) {
-            $templateName = $this->options['template'];
+        $templateName = $this->getTemplateName($model, $fallbackTemplateName);
+
+        if ($isLegacyTemplate = $this->isLegacyTemplate($templateName)) {
+            // TODO: enable deprecation once existing fragments have been adjusted
+            // trigger_deprecation('contao/core-bundle', '5.0', 'Creating fragments with legacy templates is deprecated and will not work anymore in Contao 6.');
```

Comment:
Why don‘t we adjust the existing fragments directly? At least I would enable the deprecation, so we don‘t forget about it.

Reply:
Agreed, I enabled the warning in 400216f8485edf5534f3777e089bfbde565a593e and created an [issue](https://github.com/contao/contao/issues/4461) as a reminder.

Reply:
Added the expected deprecations in f6f303dff0dcf99c118c5f0238e2392620735326. For now this is the correct way to do it.

Even when we finally create new templates in #4393 we, we will likely keep the old templates for BC (and suppressed deprecation warnings) and include the new ones from there. There are ways to achieve this, but let's please keep this out of this PR.